src/schema_compile.c

```diff
@@ -1199,6 +1197,43 @@ lys_compile_unres_llist_dflts(struct lysc_ctx *ctx, struct lysc_node_leaflist *l
     return LY_SUCCESS;
 }
 
+/**
+ * @brief Iteratively get all leafrefs from @p node
+ * if the node is of type union, otherwise just return the leafref.
+ *
+ * @param[in] node Node that may contain the leafref.
+ * @param[in,out] index Value that is passed between function calls.
+ * For each new node, initialize value of the @p index to 0, otherwise
+ * do not modify the value between calls.
+ * @return Pointer to the leafref or next leafref, otherwise NULL.
+ */
+static struct lysc_type_leafref *
+lys_type_leafref_next(const struct lysc_node *node, uint64_t *index)
+{
+    uint64_t array_count;
+    struct lysc_type *leaf_type, *union_type;
+    struct lysc_type_leafref *ret = NULL;
+
+    leaf_type = ((struct lysc_node_leaf *)node)->type;
```

Comment:
An assert should probably precede this line, that `node` is a term (`LYD_NODE_TERM`) node.